bugsnag-plugin-react-native/src/test/java/com/bugsnag/android/EventDeserializerTest.kt

```diff
@@ -115,4 +115,30 @@ class EventDeserializerTest {
         assertFalse(event.isUnhandled)
         assertTrue(TestHooks.getUnhandledOverridden(event))
     }
+
+    @Test
+    fun deserializeApiKeyOverridden() {
+        val map: MutableMap<String, Any?> = hashMapOf(
+            "apiKey" to "abc123",
+            "severity" to "info",
+            "user" to mapOf("id" to "123"),
+            "unhandled" to false,
+            "severityReason" to hashMapOf(
+                "type" to "unhandledException",
+                "unhandledOverridden" to true
+            ),
+            "breadcrumbs" to listOf(breadcrumbMap()),
+            "threads" to listOf(threadMap()),
+            "errors" to listOf(errorMap()),
+            "metadata" to metadataMap(),
+            "app" to mapOf("id" to "app-id"),
+            "device" to mapOf(
+                "id" to "device-id",
+                "runtimeVersions" to hashMapOf<String, Any>()
+            )
+        )
+
+        val event = EventDeserializer(client, emptyList()).deserialize(map)
+        assertEquals("abc123", event.apiKey)
```

Comment:
Seems worth adding an assertion when there is no API key in the map (presumably it's `null` in that case?)